### PR TITLE
Rhui reg lbrindley

### PR DESCRIPTION
**Pull request by Loquacity**
# Description

Create a new chapter in the Client Config for RHUI instructions, keep old one for CDN instructions.

# Target branches

To what product version this document applies?

- [ ] Manager-3.2
- [ ] Manager-4.0
- [x] Manager-4.1
- [x] Uyuni

# Links

Fixes https://github.com/SUSE/spacewalk/issues/11539

### CODE REVIEW

**Comment by Loquacity** [comment]
Probably more to be done here, but happy for reviews all the same :)

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 1]
```diff
@@ -0,0 +1,424 @@
+[[clients-rh-rhui]]
+= Registering {rhel} Clients with RHUI
+
+If you are running {rhel} clients directly, rather than using {sleses}, you need to use Red Hat sources to retrieve and update packages.
+This section contains information about using the {redhat} update infrastructure (RHUI) to register traditional and Salt clients running {rhel} operating systems.
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:5]: Maybe add a word about the fact it is the update method that is used for the instances that are in public clouds.

(Well, at least for Amazon EC2 - I'm not sure about the others).

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+
+For information about using {redhat} content delivery network (CDN) instead, see xref:client-configuration:clients-rh-cdn.adoc[].
+
+
+[IMPORTANT]
+====
+{rhel} clients are based on {redhat} and are unrelated to {sleses}, RES, {redhat}, or {sles}.
+You are responsible for arranging access to {redhat} base media repositories and {rhela} installation media, as well as connecting {productname} Server to the {redhat} content delivery network.
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:13]: I don't think you use installation media in the RHUI case.

Also, "{redhat} content delivery network" -> "{redhat} update infrastructure".

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+====
+
+
+
+== Import Entitlements and Certificates
+
+{redhat} clients require a {redhat} certificate authority (CA) and entitlement certificate, and an entitlement key.
+
+Entitlement certificates are embedded with expiration dates, which match the length of the support subscription.
+To avoid disruption, you will need to repeat this process at the end of every support subscription period.
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:32]: Maybe. Or maybe not. I'm not sure this is valid for RHUI.

Could you ask for second advice to someone "who knows better" please? Maybe @meaksh .

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+
+{redhat} supply a subscription manager tool to manage subscription assignments.
+It runs locally to track installed products and subscriptions.
+Clients must be registered with the subscription manager to obtain certificates.
+
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:37]: Same, I don't know if that CDN stuff also applies to RHUI.

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+====
+Satellite certificates for client systems require a Satellite server and subscription.
+Clients using Satellite certificates are not supported with {productname} Server.
+====
+
+[WARNING]
+====
+Entitlement certificates are embedded with expiration dates, which match the length of the support subscription.
+To avoid disruption, you will need to repeat this process at the end of every support subscription period.
+====
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:63]: See above.

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+
+{redhat} supplies the subscription-manager tool to manage subscription assignments.
+It runs locally on the client system to track installed products and subscriptions.
+Register to redhat.com with subscription-manager, then follow this procedure to obtain certificates.
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:67]: This time I'm almost certain this is not true.

Your machine is registered as soon as it is deployed in the cloud, there's no operation needed.

Why the "subscription-manager" utility is available, it seems unused:
```
[root@ip-172-17-17-242 ec2-user]# grep PRETTY /etc/os-release 
PRETTY_NAME="Red Hat Enterprise Linux Server 7.8 (Maipo)"

[root@ip-172-17-17-242 ec2-user]# subscription-manager status
+-------------------------------------------+
   Détails du statut du système
+-------------------------------------------+
Overall Status: Unknown

System Purpose Status: Inconnu
```

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+
+
+
+.Procedure: Registering Clients to Subscription Manager
+
+. On the client system, at the command prompt, register with the subscription manager tool:
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:73]: See above, I don't think this is needed. The beginning of this procedure can probably be nuked, and the procedure probably needs to be renamed too (it's only about copying the files now)

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
++
+----
+subscription-manager register
+----
++
+Enter your {redhat} Portal username and password when prompted.
+. Copy your entitlement certificate and key from the client system, to a location that the {productname} Server can access:
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:80]: ** Starting on this line te procedure steps are necessary. **

Remove only above.

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 9]
```diff
@@ -0,0 +1,424 @@
...
+Your entitlement certificate has a file extension of [path]``.crt``.
+For example, [path]``/etc/pki/rhui/product/content-rhel8.crt``.
+The key has a file extension of [path]``.key``.
+For example, [path]``/etc/pki/rhui/content-rhel8.key``.
+====
++
+. Copy the {redhat} CA Certificate file from the client system, to the same web location as the entitlement certificate and key:
++
+----
+cp /etc/rhsm/ca/redhat-uep.pem /example/entitlement
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:97]: er no, i think the last path is incorrect.

Here is my history of commands:
```
   12  scp /etc/pki/rhui/product/content-rhel7.crt 172.17.17.158:.
   13  scp /etc/pki/rhui/content-rhel7.key 172.17.17.158:.
   14  scp /etc/pki/rhui/cdn.redhat.com-chain.crt 172.17.17.158:.
```

File `/etc/rhsm/ca/redhat-uep.pem` exists, but I'm not sure it's of any use in the RHUI context.

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 10]
```diff
@@ -0,0 +1,424 @@
...
+* In the [guimenu]``Type`` field, select [systemitem]``SSL``.
+* In the [guimenu]``Select file to upload`` field, browse to the location where you saved the entitlement certificate, and select the [path]``.crt`` certificate file.
+. Click btn:[Create Key].
+. Click btn:[Create Stored Key/Cert] and set these parameters for the entitlement key:
+* In the [guimenu]``Description`` field, type [systemitem]``Entitlement-key-date``.
+* In the [guimenu]``Type`` field, select [systemitem]``SSL``.
+* In the [guimenu]``Select file to upload`` field, browse to the location where you saved the entitlement key, and select the [path]``.key`` key file.
+. Click btn:[Create Key].
+. Click btn:[Create Stored Key/Cert] and set these parameters for the {redhat} certificate:
+* In the [guimenu]``Description`` field, type [systemitem]``redhat-uep``.
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:120]: See above.

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 11]
```diff
@@ -0,0 +1,424 @@
...
+
+
+
+.Procedure: Adding HTTP Headers to the Configuration File
+. Locate the [systemitem]``X-RHUI-ID`` and [systemitem]``X-RHUI-SIGNATURE`` HTTP headers from your RHUI instance.
+. Open the [path]``/etc/rhn/spacewalk-repo-sync/extra_headers.conf`` configuration file, and add or edit these lines with the correct information:
++
+----
+[channel_label]
+￼X-RHUI-ID=value
```
[the remark this answers is not in the document]
Comment (Bischoff) [reply modules/client-configuration/pages/clients-rh-rhui.adoc:277]: Thanks Karl, this probably comes from my copy and paste, sorry :-) .

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 12]
```diff
@@ -0,0 +1,424 @@
...
+￼X-RHUI-SIGNATURE=value
+----
+. Alternatively, use these commands to locate the information and add it to the configuration file:
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:280]: This is correct, but might be misinterpreted as "add these commands to the configuration file", which would be wrong.

What about "Alternatively, use these commands on the RedHat instance to locate the information. Then add their result to the configuration file on the SUSE Manager Server."

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 13]
```diff
@@ -0,0 +1,424 @@
...
+
+
+[NOTE]
+====
+{rhel} channels can be very large.
+Synchronization can sometimes take several hours.
+====
+
+
+.Procedure: OPTIONAL: Creating a Salt State to Deploy Configuration Files
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:318]: Please check with @meaksh  if this is also valid in RHUI context. I did not try during my tests.

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 14]
```diff
@@ -0,0 +1,424 @@
...
+. Click btn:[Update Configuration File].
+
+
+[NOTE]
+====
+The ``Creating a Salt State to Deploy Configuration Files`` procedure is optional.
+====
+
+
+.Procedure: Creating a System Group for {rhel} Clients
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:358]: Same remark, does this apply to RHUI?

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 15]
```diff
@@ -0,0 +1,424 @@
...
+. Click btn:[Search] and the state will appear.
+. Click the checkbox for the state at the [systemitem]``Assign`` column.
+. Click btn:[Save changes].
+. Click btn:[Confirm].
+
+If you already have RHEL systems added to {productname}, assign them to the new system group, and then apply the highstate.
+
+
+
+.Procedure: Adding the System Group to Activation Keys
```
Comment (Bischoff) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:377]: Same remark again.

File: modules/client-configuration/pages/clients-rh-rhui.adoc (hunk 3 above)
Comment (Loquacity) [reply modules/client-configuration/pages/clients-rh-rhui.adoc:32]: Sure. @meaksh ?

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 16]
```diff
@@ -0,0 +1,424 @@
...
+To manage repositories on your {redhat} client, you need to import the CA and entitlement certificates to the {productname} Server.
+This requires that you perform the import procedure three times, to create three entries: one each for the entitlement certificate, the entitlement key, and the {redhat} certificate.
+
+
+
+.Procedure: Importing Certificates to the Server
+
+. On the {productname} Server {webui}, navigate to menu:Systems[Autoinstallation > GPG and SSL Keys].
+. Click btn:[Create Stored Key/Cert] and set these parameters for the entitlement certificate:
+* In the [guimenu]``Description`` field, type [systemitem]``Entitlement-Cert-date``.
```
[the remark this answers is not in the document]
Comment (Loquacity) [reply modules/client-configuration/pages/clients-rh-rhui.adoc:110]: it's just a description, so it shouldn't really matter

File: modules/client-configuration/pages/clients-rh-rhui.adoc (hunk 12 above)
Comment (Loquacity) [reply modules/client-configuration/pages/clients-rh-rhui.adoc:280]: OK, will rework this procedure.

File: modules/client-configuration/pages/clients-rh-rhui.adoc (hunk 13 above)
Comment (meaksh) [reply modules/client-configuration/pages/clients-rh-rhui.adoc:316]: This does not make sense on the context of RHUI. I think we should drop this procedure.

File: modules/client-configuration/pages/clients-rh-rhui.adoc (hunk 14 above)
Comment (meaksh) [reply modules/client-configuration/pages/clients-rh-rhui.adoc:356]: This one does not make sense on the RHUI context.

File: modules/client-configuration/pages/clients-rh-rhui.adoc (hunk 15 above)
Comment (meaksh) [reply modules/client-configuration/pages/clients-rh-rhui.adoc:375]: This one does not make sense on the RHUI context.

File: modules/client-configuration/pages/clients-rh-rhui.adoc (hunk 3 above)
Comment (meaksh) [reply modules/client-configuration/pages/clients-rh-rhui.adoc:41]: This seems to be a general note about how susbscriptions works on the RHEL world, particularly helpful when dealing directly with the CDN, where the entitlements you are related to your own subscription. So you need to take care of update the entitlement every time your subscription expires.

But for RHUI, the entitlement are not yours but owned by the cloud provider, so you don't really know when they expired. Maybe we should avoid this particular text on the RHUI guide.

File: modules/client-configuration/pages/clients-rh-rhui.adoc (hunk 4 above)
Comment (meaksh) [reply modules/client-configuration/pages/clients-rh-rhui.adoc:46]: This part about  "subscription manager" is only valid on CDN guide, it does not make sense here.

File: modules/client-configuration/pages/clients-rh-rhui.adoc (hunk 5 above)
Comment (meaksh) [reply modules/client-configuration/pages/clients-rh-rhui.adoc:72]: Same consideration as in the above comment about the entitlement.

File: modules/client-configuration/pages/clients-rh-rhui.adoc [hunk 17]
```diff
@@ -62,39 +71,27 @@ Entitlement certificates are embedded with expiration dates, which match the len
...
-{redhat} supplies the subscription-manager tool to manage subscription assignments.
-It runs locally on the client system to track installed products and subscriptions.
-Register to redhat.com with subscription-manager, then follow this procedure to obtain certificates.
 
 
+The entitlement certificates and keys need to be copied from the client system to a location that the {productname} Server can access.
 
-.Procedure: Registering Clients to Subscription Manager
+Your entitlement certificate and the {redhat} CA Certificate file have file extensions of [path]``.crt``.
+The key has a file extension of [path]``.key``..
```
Comment (keichwa) [inline modules/client-configuration/pages/clients-rh-rhui.adoc:79]:
```suggestion
The key has a file extension of [path]``.key``.
```

**Pull request merged**